packages/blade/src/components/Tag/_decisions/decisions.md

```diff
@@ -0,0 +1,198 @@
+
+```jsx
+<TextInput tagsSlot={} value="" onChange={} />
+```
+
+##### Alternate Prop Name
+
+- `leading` (felt confusing because it is leading to the invisible input tag and not the actual visible TextInput)
+- `valuePrefix` (Similar to `titleSuffix` from BaseHeader)
+
+<details>
+<summary>Full Controlled Example</summary>
+
+```jsx
+function App() {
+  const [inputValue, setInputValue] = React.useState('');
+  const [tags, setTags] = React.useState([]);
+
+  const addTag = () => {
+    // Add input value to tags and clear the input value
+    setTags([...tags, inputValue]);
+    setInputValue('');
+  }
+
+  const removeTag = (tagName) => {
+    setTags(tags.filter(tagNameValue) => tagNameValue !== tagName);
+  }
+
+  return (
+    <TextInput
+      tagsSlot={tags.map((tagName, index) => (
+        <Tag onDismiss={({ value }) => removeTag(value)}>{tagName}</Tag>
+      ))}
+      value={inputValue}
+      onChange={({ value }) => setInputValue(value)}
+      onKeyDown={(e) => {
+        if (e.key === 'ENTER') {
+          addTag();
+        }
+      }}
+    />
+  );
+}
+```
+
+</details>
+
+<details>
+<summary>Alternate Approach</summary>
+
+#### Alternate Approach: `value` prop
+
+We can extend our `value` prop to accept JSX
+
+```jsx
+<TextInput
+  value={
+    <>
+      <Tag onDismiss={}>kamlesh.chandnani@razorpay.com</Tag>
+      <Tag onDismiss={}>divyanshu.maithani@razopay.com</Tag>
+      saurabhdaw
+    </>
+  }
+  onChange={}
+/>
+```
+
+**Cons**
+
+- Requires us to loop on children and separate out Tags from written word and this will run on every `onChange` event.
+- Have not seen any other library handle it this way
+
+</details>
+
+## Accessibility
+
+- Ensure Close Icon is focussable
+- Close Icon to have appropriate `aria-label`
+- Special keyboard interaction when used inside Input components
+  - Pressing `Tab` should move focus into the input and not Tag
+  - Tag is focussable with Left and Right Arrow Keys
+  - Pressing `Backspace` on Tag should remove it
+  - Clicking on Tag should bring focus on Input
+- Tag should be disabled when used inside a disabled input
+
+Refer to Keyboard Interactions of [react-select](https://react-select.com/home#getting-started) for reference. Other Input components with Tag also follow similar keyboard interactions.
+
+## References
+
+- [react-select](https://react-select.com/home#getting-started)
+- [`TagsInput` component by PluralSight Design System](https://design-system.pluralsight.com/components/tagsinput)
+- [`tokens` prop by Primer AutoComplete.Input](https://primer.style/react/Autocomplete/)
+- [Carbon just putting everything outside of Inputs](https://carbondesignsystem.com/components/tag/usage)
+- [`elemBeforeInput` prop on Atlassian TextField](https://atlassian.design/components/textfield/examples#elements-before-and-after-input)
+
+## Open Questions
+
+- Do we want to discuss Tag's integration with TextInput and TextArea? or should I remove it from API decision altogether?
```

Comment:
If we add it on BaseTextInput only then its an implementation detail and we can remove it from here

Reply:
It won't be just on BaseTextInput. It will also be publically available on TextInput and TextArea eventually since we have usecases where we want to enter email ids on TextArea which become tags as we type